packages/api/src/models/animalModel.js

```diff
@@ -24,6 +25,28 @@ class Animal extends baseModel {
     return 'id';
   }
 
+  async $beforeInsert(queryContext) {
+    await super.$beforeInsert(queryContext);
+    this.trimStringProperties();
+  }
+
+  async $beforeUpdate(opt, queryContext) {
+    await super.$beforeUpdate(opt, queryContext);
+    this.trimStringProperties();
+  }
+
```

Comment:
I added this here because:

1) There were too many string properties on this table to check individually in the controller
2) String trimming had not been done on POST. `addAnimals()` is already a pretty length controller so I did not want to lengthen it

Curious if people like this pattern enough to generalize it; for now it is just on these two models.